### ISSUE TITLE
Missing polyfills for older browsers, especially bind, requestAnimationFrame.

### ISSUE THREAD

**Issue by jandrieu** [dmvaldman/samsara#41]
In Qt 4.8/QtWebKit 2.2, both bind() and requestAnimationFrame() fail. I can't verify which version of QtWebKit (or its successor QtWebEngine) start providing bind, but I do know the old Phantom.js also had this problem. I believe Qt5.3 and Phantom2 both support bind, but I don't know when requestAnimationFrame gets supported.

We could either document how to add known good polyfills with `<script>` elements, or provide working polyfills within Samsara.

**Comment by culyun** [comment]
I'm curious, what's your use case for Qt4.8 Webkit?

**Comment by jandrieu** [comment]
@culyun That's what we have on our current platform. We started migrating to 5.4, but it isn't well supported with the existing graphics SDK and toolchain. We upgrade the SDK a few notches, but still had problems. Then the budget got turned to higher priotity issues before we could work through those issues. So, for the foreseeable future, we're stuck with Qt4.8. and Qt2.2. FWIW, we did look into Qt-WebKit 2.3 - a backport that provides several Qt 5.4 features to 4.8 -- WebGL and bind() and probably requestAnimationFrame -- but that had dependency issues that ended up looking more complicated than pushing to 5.4 and QtWebEngine.

Unfortunately, our ARM SOC has a relatively narrow window of support wrt toolchains, OpenGL, Qt, etc. Pushing beyond the tech that was supported when it first shipped can be a significant effort. My experience is that this is typical for embedded platforms.

**Comment by jandrieu** [comment]
I closed the pull request #42. Is there a wiki or somewhere I could write something up about the errors I hit and how to use polyflls as a work around?

**Comment by dmvaldman** [comment]
A wiki is a good idea. I'll think about a good way to organize it!

**Comment by culyun** [comment]
@jandrieu you're not fighting Windows CE per chance?  I have quite a bit of experience with Qt 4 and 5 on Wince.

**Comment by jandrieu** [comment]
@culyun We're on embedded linux. It looks like 5.4 would give us some huge advantages, including rAF... but sadly that's just not in the cards for the foreseeable future.  With the rAF polyfill, we're getting maybe 1-3fps and we're slamming the CPU (70%+). =(  The hardware can do better--the OpenGL examples I've tried scream at 20+fps and 5% CPU, but without the upgrade to 5.4, we're fighting an uphill battle.  But at least I was able to get it to run.
